### PR TITLE
petsc: update to version 3.14.4, and petsc4py to 3.14.1

### PR DESCRIPTION
**Pull request by balay**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-petsc4py/package.py [hunk 1]
```diff
@@ -42,7 +43,8 @@ class PyPetsc4py(PythonPackage):
     depends_on('petsc+mpi', when='+mpi')
     depends_on('petsc~mpi', when='~mpi')
     depends_on('petsc@develop', when='@develop')
-    depends_on('petsc@3.14:3.14.99', when='@3.14:3.14.99')
+    depends_on('petsc@3.14.2:3.14.99', when='@3.14.2:3.14.99')
+    depends_on('petsc@3.14:3.14.1', when='@3.14')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-petsc4py/package.py]: Should this be:
```suggestion
    depends_on('petsc@3.14:3.14.1', when='@3.14.0:3.14.1')
```
I can never remember how shortened versions work.

Comment (balay) [reply var/spack/repos/builtin/packages/py-petsc4py/package.py]: Ah - my mistake. Pushed a fix.

File: var/spack/repos/builtin/packages/py-petsc4py/package.py [hunk 2]
```diff
@@ -42,7 +43,8 @@ class PyPetsc4py(PythonPackage):
     depends_on('petsc+mpi', when='+mpi')
     depends_on('petsc~mpi', when='~mpi')
     depends_on('petsc@develop', when='@develop')
-    depends_on('petsc@3.14:3.14.99', when='@3.14:3.14.99')
+    depends_on('petsc@3.14.2:3.14.99', when='@3.14.1:3.14.99')
+    depends_on('petsc@3.14.0:3.14.1', when='@3.14.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-petsc4py/package.py:47]: Just making sure, `py-petsc4py@3.14.1` depends on `petsc@3.14.2:`, not `petsc@3.14.1:`?

Comment (balay) [reply var/spack/repos/builtin/packages/py-petsc4py/package.py:47]: yes.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**